astropy/units/format/__init__.py

```diff
@@ -43,6 +50,22 @@
 ]
 
 
+@runtime_checkable
+class UnitFormatter(Protocol):
+    @classmethod
+    @abstractmethod
+    def to_string(cls, unit: UnitBase) -> str:
+        raise NotImplementedError
+
+
+@runtime_checkable
+class ParsingUnitFormatter(UnitFormatter, Protocol):
```

Comment:
Does this need to inherit from `Protocol` on its own (rather than just through `UnitFormatter`?

Reply:
A short summary of the [Explicitly declaring implementation](https://peps.python.org/pep-0544/#explicitly-declaring-implementation) and [Merging and extending protocols](https://peps.python.org/pep-0544/#merging-and-extending-protocols) sections of PEP 544 is that a protocol can provide default implementations of its methods that normal classes can adopt by explicitly inheriting from the protocol, which means that explicit subclasses of a protocol cannot be assumed to be protocols themselves.

Reply:
Thanks, that helps. Would it be an idea for the protocol to then define the default `.parse()` implementation? (And then have the general classes inherit from it?)